tickets/SP-1635

MelissaGraham opened the pull request:
Add new page summarizing RSP user resources (storage, compute, etc.).

jonathansick commented on docs/guides/account-resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+
+- Notebook Aspect (JupyterLab servers): 1 :sup:`b`
+- Portal Aspect sessions: 1 :sup:`b`
+- API connections: TBD
```
FYI, the API connections for a user will be based on rate-limit quotas (e.g. requests per amount of time).

MelissaGraham replied:
Added a new footnote to specify that.

jonathansick reviewed (commented):
First, could you name the PR? I'd recommend: `SP-1635: Document account resources`. This makes it easier for the RSP team.

I'd recommend shortening the URL a bit: `docs/guide/resources`.

Another thing to consider is separating what resources are given to a user from the overall platform capacity. Right now both are stated in the same sections. The information actionable to a user is their own quotas, and I think that information should come first. The platform capacity and estimates of how many users can be on the platform before it degrades can be a later reference section because it's not action-able information to our users.

Lastly, I notice a lot of intended footnotes in this page. Here's the actual syntax for creating footnotes: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#footnotes The native syntax links between the footnote marker and the footnote, so it's useful.

MelissaGraham commented:
Thanks @jonathansick. Agree on all suggestions and incorporated them all in one commit.

 - changed PR title to "SP-1635: page for per-account user compute resources"
 - rebased to main
 - yes I can squash-and-merge the PR when the time comes
 - renamed folder as /guides/resources
 - moved "Number of users" sub-section down to the bottom

I'd kind of wanted the "footnotes" to be within each subsection but I switched to using the RST format for bottom-of-the-page footnotes.

MelissaGraham commented:
Not merging and not requesting review because this page is not finished, still awaiting inputs and final quantity estimates.

womullan reviewed (commented):
ok for me - TBDs need SQuaRE and USDF to comment

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+Notebook server options:
+
+- Small (1.0 CPU, 4 GB RAM)
+- Medium (2.0 CPU, 8 GB RAM)
```
Sorry I keep neglecting to bring this up, but I am severely tempted to remove medium as an option. It seems almost nobody is using it and it risks confusing users who try to do things that run out of memory. Discuss this on slack?

(Also I might slightly tweak those values to better utilize those nodes but no worries, we can change the numbers later)

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+.. important::
+   The US DAC, the Rubin Science Platform, and this documentation are currently under construction.
+   The estimates quoted below include current, planned, and to-be-determined (TBD) values, all of which are subject to change as hardware, software, and user habits evolve.
+
```
I don't know whether it would be useful here to link to the roadmap and say additional services are planned. Up to you.

MelissaGraham replied:
Added "See the Roadmap for planned changes (top menu bar).".

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+Only CPUs (central processing units) are available.
+No GPU (graphical processing units) are available and there is no plan to add them.
+
+- Cores per science user: 0.1 [#f1]_
```
sorry what does this mean

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+- Large (4.0 CPU, 16 GB RAM)
+
+Only CPUs (central processing units) are available.
+No GPU (graphical processing units) are available and there is no plan to add them.
```
```suggestion
No GPU (graphical processing units) are available and there is no immediate plan to add them, though we are evaluating ways to make them available to the community alongside the RSP. 
```

MelissaGraham replied:
Added immediate, but didn't add that the option is being evaluated, no need to tell users that unless they're definitely coming soon.

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+
+During the Early Science era (the Data Previews and Data Release 1), the total amount of shared user space across all user directories (home, project, and scratch) is 24 TB.
+
+During the Operations era, the anticipated amount of individual home directory disk space (in the cloud) is 0.4 TB [#f3]_.
```
That's not what's in the sizing model, I don't think. I will double check.  But I believe it's factor of 10 off and that's only assuming 5,000 users. Let me provide some wording for this section.

MelissaGraham replied:
Alright I'll leave this one unresolved until we have new wording thanks

frossie replied:
I would like to stick to the wording in the text I wrote here if that is okay. 
https://github.com/lsst/rsp_lsst_io/pull/44/files

MelissaGraham replied:
Sure -- incorporated that text into "Storage" and new subsection "Scratch".

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+Storage
+=======
+
+During the Early Science era (the Data Previews and Data Release 1), the total amount of shared user space across all user directories (home, project, and scratch) is 24 TB.
```
I don't think that's right and also I am not sure giving total space here is useful anyway.

MelissaGraham replied:
Alright I'll leave this one unresolved until we can confirm the value, or if you decide it shouldn't be quoted at all. It would be useful in the sense of informative and preempting user questions, even though it's not a limit that would affect individual users.

frossie replied:
let's leave it out. There's no way to give a meaningful number (because it will depend on our overcommitment rate)

MelissaGraham replied:
Super -- and it's already been replaced as a result of updates to the "Storage" section made for another comment.

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+
+Users are encouraged to use services such as GitHub for software version control and to take care not to accidentally delete files from their home directory.
+
+While there is no guarantee that accidentally deleted data can be recovered, users are encouraged to use the resources for :doc:`/support/index` immediately if mistakes do happen.
```
Yeah I don't know what the right story here is. We don't really have the resources to be randomly recovering "oops I deleted foo.txt" for users and if this ended up being a thing I'd have to build a system to allow them to self-serve. I would be tempted to say "Backup mechanisms exist to project your data in case of system failures" and leave it at that.

MelissaGraham replied:
The sentence is now "While there is no guarantee that accidentally deleted data can be recovered, backup mechanisms do exist to protect user data in case of system failures.". No mention of asking for help to recover accidentally deleted files.

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+
+All queries are executed with shared resources.
+The length of time to query completion depends firstly on query design (number of shards accessed), and secondly on number of queries across all users.
+There is no limit on the number of queries a user can do in total (or on daily or yearly timescales), but there are query rate limits.
```
I may be confused but isn't a query rate limit equivalent to the number of queries you can effectively do in a day? I think it might be clearer to drop the parenthetical content

MelissaGraham replied:
Deleted " (or on daily or yearly timescales)".

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+- Reset interval after user excession: 15 minutes [#f5]_
+
+
+Portal Aspect TAP (Table Access Protocol) service:
```
We were discussing this today and I will open an RFD to get you final values

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+
+Number of simultaneous users above which service may degrade:
+
+- Notebook Aspect (JupyterLab servers): 517 [#f6]_
```
```suggestion
- Notebook Aspect (JupyterLab servers): 3,000 [#f6]_
```

MelissaGraham replied:
Made this change.

frossie commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+- Maximum rows returned: 5,000,000
+- Maximum table size returned: TBD
+
+Download and upload limits
```
This is another we-don't-have-a-great-story here, since the RSP does not offer bulk download, and there are different answers for Internet 2 and "public" Internet sites, and also we don't have a recommended mechanism, nor have we estimated throttle values for eg image APIs. If you want this document to go out sooner rather than later you might want to omit this section for now.

MelissaGraham replied:
Left the section header and explanatory text, deleted all the "TBD" rows.

frossie reviewed (commented):
I owe you words for the storage section still. Working on these now. Apologies for the delay.

MelissaGraham commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+It is a requirement defined by the `Science Requirements Document <https://docushare.lsst.org/docushare/dsweb/Get/LPM-17>`_ that the "Data Management System will also provide at least 10% of its total capability for user-dedicated processing and user-dedicated storage" (Section 3.5).
```
Deleted that first line - it's not necessary.

MelissaGraham commented on docs/guides/resources/user-account-resources.primary.in.rst:
```diff
@@ -0,0 +1,168 @@
+This page quantifies the individual and shared resources that every account at the US Data Access Center (US DAC) has access to, by default, via the Rubin Science Platform at |rsp-url|.
+
+.. important::
+   The US DAC, the Rubin Science Platform, and this documentation are currently under construction.
```
Added "The Rubin Science Platform, and the values on this page, will continue to evolve through Rubin Operations.".
